main.py

```diff
@@ -1,87 +1,156 @@
+#     current_wattages = []
+#     client.write_points(json_body)
+
+
+class DmsrFourTelegram:
+    current_usage_electricity: float
+    total_usage_electricity_low: float
+    total_usage_electricity_high: float
+
+@dataclass
+class DsmrInfo:
+    variable_name: str
+    search_for: str
+    unit: str
+
 
+class DsmrFour():
+    _serial : serial.Serial
+    lines: List[str] = []
+    ending_character: str = "!"
 
+    _registrations: List[DsmrInfo] = []
+    def __init__(self, ser: serial.Serial):
+        ser.baudrate = 115200
+        ser.bytesize =serial.EIGHTBITS
+        ser.parity = serial.PARITY_NONE
+        ser.stopbits = serial.STOPBITS_ONE
+        ser.xonxoff = 0
+        ser.rtscts = 0
+        ser.timeout = 12
+        ser.port = "/dev/ttyUSB0"
+        self._serial = ser
     
-  
- 
-    # Check wanneer het uitroepteken ontvangen wordt (einde telegram)
-    if re.match('(?=!)', telegram_line):
-      checksum_found = True
-      add_wattage(watt)
- 
-  ser.close()
+    def register(self,data_unit: DsmrInfo)  -> None:
+        self._registrations.append(data_unit)
+
+    def interpret(self, lines: List[str]) -> List:
+        data = []
+        for finder in self._registrations:
+            for text in lines:
+                if finder.search_for in text:
+                    groups = re.findall(rf'([0-9]+[\.]+[0-9]*\*{finder.unit})',text)
+
+                    text = [group.replace(f"*{finder.unit}", "") for group in groups if str(group).endswith(f"*{finder.unit}")][0]
+                    actual_float = float(text)
+                    data.append({
+                        'variable_name' : finder.variable_name,
+                        'value' : actual_float,
+                        'unit' : finder.unit
+                    })
+        return data
+
+    def get_lines(self) -> List[str]:
+        return self.lines
+
+    def print_lines(self) -> None:
+        for line in self.lines:
+            print(line)
+
+    def end_of_telegram(self, line: str) -> bool:
+        return self.ending_character in line
+
+
+    def wait_for_telegram_and_load(self) -> None:
+        self._serial.open()
+        self.lines = []
+        while  True:
+            telegram_line = self._serial.readline()
+            telegram_line = telegram_line.decode('ascii').strip()
+            if self.end_of_telegram(telegram_line):
+                break
+            else:
+                self.lines.append(telegram_line)
+
+        self._serial.close()
+
+
+dsmr = DsmrFour(serial.Serial())
+data_units = [
+    DsmrInfo(variable_name="total_gas", search_for="0-1:24.2.1", unit="m3"),
+    DsmrInfo(variable_name="current_electricity", search_for="1-0:1.7.0",  unit="kW"),
+    DsmrInfo(variable_name="total_electricity_high", search_for="1-0:1.8.1", unit="kWh"),
+    DsmrInfo(variable_name="total_electricity_low", search_for="1-0:1.8.2", unit="kWh"),
+]
+
+for du in data_units:
+    dsmr.register(du)
+
+while True:
+    dsmr.wait_for_telegram_and_load()
```

Comment:
Is there any use case where you don't do these three steps consecutively? Otherwise, it might make sense to just add a method where you call these three and return the end result. Currently you need to know about implementation details of the class.